rust/driver_manager/src/lib.rs

```diff
@@ -382,7 +385,23 @@ impl ManagedDriver {
 
         let entrypoint = entrypoint.unwrap_or(default_entrypoint.as_bytes());
         let library = unsafe {
-            libloading::Library::new(filename.as_ref()).map_err(libloading_error_to_adbc_error)?
+            // By default, go builds the libraries with '-Wl -z nodelete' which does not
+            // unload the go runtime. This isn't respected on mac ( https://github.com/golang/go/issues/11100#issuecomment-932638093 )
+            // so we need to explicitly load the library with RTLD_NODELETE( which prevents unloading )
+            if cfg!(target_os = "macos") {
```

Comment:
I suppose the cfg attribute `#[cfg(...)]` is better than `cfg!` here.
(In this case, it will compile on any OS.)

Reply:
Makes sense. Updated the code to use `#[cfg(...)]`